### PR TITLE
Bump version of payloadoffloading-common library and code refactor for SDKv2

### PR DESCRIPTION
**Pull request by agummi-amazon**
*Issue #7  :*

*Description of changes:*
Bumped payloadoffloading-common library version to v2.1.2
Refactored code to work with SDKv2

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -94,7 +94,7 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-compiler-plugin</artifactId>
-                    <version>3.8.1</version>
+                    <version>3.2</version>
```
Comment (shih) [inline pom.xml:97]: Why downgrade?

File: src/main/java/software/amazon/sns/AmazonSNSExtendedClient.java [hunk 2]
```diff
@@ -19,6 +19,7 @@
 
 public class AmazonSNSExtendedClient extends AmazonSNSExtendedClientBase {
     static final String MULTIPLE_PROTOCOL_MESSAGE_STRUCTURE = "json";
+    private static final String S3_KEY = "S3Key";
```
Comment (shih) [inline src/main/java/software/amazon/sns/AmazonSNSExtendedClient.java:22]: This should be put near line 26 as they are both `private static final`

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "java.configuration.updateBuildConfiguration": "automatic"
+}
```
Comment (mangeche) [inline .vscode/settings.json]: This file is removed from https://github.com/awslabs/amazon-sns-java-extended-client-lib/pull/17/commits/541d253746a36622e0945c149fae7810950aeaf8 and added back here. Can you combine all commits together?

File: pom.xml [hunk 4]
```diff
@@ -120,7 +120,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.0.1</version>
+                <version>2.9.1</version>
```
Comment (mangeche) [inline pom.xml:123]: Same as above

File: pom.xml (hunk 1 above)
Comment (agummi-amazon) [reply pom.xml:97]: Using the same version used in java-sqs-extended-client.

Comment (shih) [reply pom.xml:97]: It is not necessary to have the same version, please revert.

File: pom.xml (hunk 4 above)
Comment (agummi-amazon) [reply pom.xml:123]: The documentation format for the sns-extended-client works best with javadoc v2.9.1 which was being used for generating the documentation for the prior 1.x versions of the extended client. Locally I was developing on Maven3 which had compatablity issues with v2.9.1 so I had upgraded the version but with maven2 which has been used before to deploy the project before v2.9.1 is compatible.

If it needs to be upped again pls inform.

File: .vscode/settings.json (hunk 3 above)
Comment (agummi-amazon) [reply .vscode/settings.json]: Yes will squash all commits together.

File: src/main/java/software/amazon/sns/AmazonSNSExtendedClient.java (hunk 2 above)
Comment (agummi-amazon) [reply src/main/java/software/amazon/sns/AmazonSNSExtendedClient.java]: changed.

File: pom.xml (hunk 1 above)
Comment (agummi-amazon) [reply pom.xml]: reverted to v3.8.1

**Review by shih — APPROVED** [review]

**Pull request merged**